lib/bolt/pal/yaml_plan/transpiler.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+require 'bolt/pal/yaml_plan/loader'
+require 'bolt/util'
+
+module Bolt
+  class PAL
+    class YamlPlan
+      class Transpiler
+        class ConvertError < Bolt::Error
+          def initialize(msg, plan_path)
+            super(msg, 'bolt/convert-error', { "plan_path" => plan_path })
+          end
+        end
+
+        def transpile(relative_path)
+          @plan_path = File.expand_path(relative_path)
+          @modulename = Bolt::Util.module_name(@plan_path)
+          @filename = @plan_path.split(File::SEPARATOR)[-1]
+          validate_path
+
+          plan_object = parse_plan
+
+          plan_string = String.new("# WARNING: This is an autogenerated plan. " \
+                                   "It may not behave as expected.\n" \
+                                   "plan #{plan_object.name}(")
+          # Parameters are Bolt::PAL::YamlPlan::Parameter
+          plan_object.parameters&.each_with_index do |param, i|
+            plan_string << param.transpile
+
+            # If it's the last parameter add a newline and no comma
+            last = i + 1 == plan_object.parameters.length ? "\n" : ","
+            # This encodes strangely if we << directly to plan_string
+            plan_string << last
+          end
+          plan_string << ") {\n"
+
+          plan_object.steps&.each do |step|
+            # This only needs the plan path for raising errors
+            plan_string << step.transpile(@plan_path)
+          end
+
+          plan_string << "\n  return #{Bolt::Util.to_code(plan_object.return)}\n" if plan_object.return
+          plan_string << "}"
+          # We always print the plan, even if there's an error
+          puts plan_string
+          validate_plan(plan_string)
+          plan_string
+        end
+
+        def parse_plan
+          file_contents = File.read(@plan_path)
```

Comment:
This will stack trace if path doesn't exists.

Reply:
haha this function is 100% begin..rescues now....